ide/che-core-ide-generators/src/main/java/org/eclipse/che/util/GwtXmlGenerator.java

```diff
@@ -0,0 +1,259 @@
+            this.excludePackages = excludePackages;
+            this.includePackages = includePackages;
+            this.urls = urls;
+        }
+
+        /**
+         * Searches XXX.gwt.xml files in class path
+         *
+         * @return - set of XXX.gwt.xml files found in class path
+         */
+        public Set<String> getGwtModulesFromClassPath() {
+            ConfigurationBuilder configurationBuilder = getConfigurationBuilder();
+            if (urls != null && urls.size() > 0) {
+                configurationBuilder.addUrls(urls);
+            }
+            FilterBuilder filterBuilder = new FilterBuilder();
+            for (String includePackage : includePackages) {
+                filterBuilder.includePackage(includePackage);
+            }
+            for (String excludePackage : excludePackages) {
+                filterBuilder.excludePackage(excludePackage);
+            }
+
+
+            configurationBuilder.setScanners(new ResourcesScanner()).filterInputsBy(filterBuilder);
+
+            Reflections reflection = new Reflections(configurationBuilder);
+            return new TreeSet<>(reflection.getResources(name -> name.endsWith(".gwt.xml")));
+
+        }
+    }
+
+    public static class GwtXmlGeneratorConfig {
+        private final Set<String> gwtXmlModules;
+
+        private final File generationRoot;
+
+        private final String gwtFileName;
+
+        private final String entryPoint;
+
+        private final String stylesheet;
+
+        private final boolean isLoggingEnabled;
+
+        public GwtXmlGeneratorConfig(Set<String> gwtXmlModules,
+                                     File generationRoot,
+                                     String gwtFileName,
+                                     String entryPoint,
+                                     String stylesheet,
+                                     boolean isLoggingEnabled) {
+            this.gwtXmlModules = gwtXmlModules;
+            this.generationRoot = generationRoot;
+            this.gwtFileName = gwtFileName;
+            this.entryPoint = entryPoint;
+            this.stylesheet = stylesheet;
+            this.isLoggingEnabled = isLoggingEnabled;
+        }
+
+        public GwtXmlGeneratorConfig(Set<String> gwtXmlModules,
+                                     File generationRoot) {
+            this(gwtXmlModules,
+                 generationRoot,
+                 DEFAULT_GWT_XML_PATH,
+                 DEFAULT_GWT_ETNRY_POINT,
+                 DEFAULT_STYLE_SHEET,
+                 DEFAULT_LOGGING);
+        }
+        
+        public Set<String> getGwtXmlModules() {
+            return gwtXmlModules;
+        }
+
+        /**
+         * @return set of gwt modules where '/' replaced with '.' and removed 'gwt.xml' from end.
+         * used to insert as inherits in gwt.xml
+         */
+        public Set<String> getGwtModules() {
```

Comment:
Now it's much clear :) But I'd say that `getGwtModules` - `getGwtModulesFiles`, and `getGwtModules` - `getGwtModulesNames`